fdbserver/ClusterController.actor.cpp

```diff
@@ -478,11 +505,30 @@ class ClusterControllerData {
 			auto fitness = worker_details.processClass.machineClassFitness(ProcessClass::TLog);
 
 			if (std::find(exclusionWorkerIds.begin(), exclusionWorkerIds.end(), worker_details.interf.id()) !=
-			        exclusionWorkerIds.end() ||
-			    !workerAvailable(worker_info, checkStable) ||
-			    conf.isExcludedServer(worker_details.interf.addresses()) || fitness == ProcessClass::NeverAssign ||
-			    (!dcIds.empty() && dcIds.count(worker_details.interf.locality.dcId()) == 0) ||
-			    (!allowDegraded && worker_details.degraded)) {
+			    exclusionWorkerIds.end()) {
+				logWorkerUnavailable(id, "complex", "Worker is excluded", worker_details, fitness, dcIds);
+				continue;
+			}
+			if (!workerAvailable(worker_info, checkStable)) {
+				logWorkerUnavailable(id, "complex", "Worker is not available", worker_details, fitness, dcIds);
+				continue;
+			}
+			if (conf.isExcludedServer(worker_details.interf.addresses())) {
+				logWorkerUnavailable(
+				    id, "complex", "Worker server is excluded from the cluster", worker_details, fitness, dcIds);
+				continue;
+			}
+			if (fitness == ProcessClass::NeverAssign) {
+				logWorkerUnavailable(id, "complex", "Worker's fitness is NeverAssign", worker_details, fitness, dcIds);
+				continue;
+			}
+			if (!dcIds.empty() && dcIds.count(worker_details.interf.locality.dcId()) == 0) {
+				logWorkerUnavailable(id, "complex", "Worker is not in the target DC", worker_details, fitness, dcIds);
+				continue;
+			}
```

Comment:
Can you make these two debug traces (trace event will be logged only when debug tracing is enabled), here and below?

I find that these two cases are easy to figure out, and currently they generate quite a bit of noise in the trace event: it is expected that there will always be some workers that are not in the target dc, and have wrong fitness.

Reply:
Done.